### ISSUE TITLE
Add ability to specify maximum message size

### ISSUE THREAD

**Issue by acogoluegnes** [rabbitmq/rabbitmq-java-client#1062]

**Comment by gdimitrov7** [comment]
The [releases list](https://github.com/rabbitmq/rabbitmq-java-client/releases) and the commit history both show that the fix was backported to: 
[v5.17.1](https://github.com/rabbitmq/rabbitmq-java-client/releases/tag/v5.17.1)
[v5.16.1](https://github.com/rabbitmq/rabbitmq-java-client/releases/tag/v5.16.1)
[v5.14.3](https://github.com/rabbitmq/rabbitmq-java-client/releases/tag/v5.14.3)

@acogoluegnes Can you confirm?

I want confirmation because 

- https://nvd.nist.gov/vuln/detail/CVE-2023-46120#vulnConfigurationsArea
- https://github.com/rabbitmq/rabbitmq-java-client/security/advisories/GHSA-mm8h-8587-p46h

show that the fix is only available in 5.18.0+ which is only one branch. 

@michaelklishin Is it possible to update the security advisories with correct versions? 
Thank you.

**Comment by michaelklishin** [comment]
@gdimitrov7 you can inspect the branches of those releases just like we can, [here is `5.17.x-stable`](https://github.com/rabbitmq/rabbitmq-java-client/commits/5.17.x-stable), for instance.

GitHub security advisory only has one "affected versions" field IIRC. There are no reasons to to prefer 5.17 over the latest 5.x, this client ships minors often and usually without breaking changes of any kind.

**Comment by michaelklishin** [comment]
@gdimitrov7 we cannot retroactively update NIST database entries, therefore we won't update the advisory even if there was/is a way to list multiple affected versions.

**Comment by acogoluegnes** [comment]
@gdimitrov7 Yes, it's been backported to those patch releases. You should use 5.20.0 (the latest as of today) though, it's backward compatible.

**Comment by peterwvz** [comment]
> we cannot retroactively update NIST database entries, therefore we won't update the advisory even if there was/is a way to list multiple affected versions.

You can actually update NVD. Happens all the time. Reanalysis efforts are typically driven by external engagement such as emails to [nvd@nist.gov](mailto:nvd@nist.gov) or [cpe_dictionary@nist.gov](mailto:cpe_dictionary@nist.gov). You'd include the pertinent info (like this thread) showing the backports happened. I think that's the only way to get all the enterprise vulnerability scanners to acknowledge that 5.18.0 is not the only fix for this CVE.

**Comment by michaelklishin** [comment]
There are also no reasons for the 5.17 users not to upgrade to 5.18 or 5.20. As stated earlier, this client usually ships minor releases without any breaking changes.

Our small team has more important things to worry about than making sure a three minor behind patch release of this client has a backport accounted for in the NIST database.

We have fixed the issue and released a round of back ports. NIST is notified. If someone has the cycle to update the NIST database, they can do it by providing evidence from this public repository.

**Comment by peterwvz** [comment]
Understood. I am working with a third party vendor. They are on 5.14.3 and
getting caught by an enterprise vuln. scanner flagging that versions as
vulnerable - saying that they need to upgrade to 5.18.0.

> this client usually ships minor releases without any breaking changes.
It's the "usually" in that sentence that is the problem. The vendor has
other critical CVEs that need addressing. And upgrading any component
involves rerunning all the regressions tests/acceptance test, etc. etc.
Work they'd prefer to avoid and they know they can avoid since they
actually do have the backported fix!! The only thing that doesn't
understand that is the enterprise vuln. scanners....

pete




On Mon, Feb 19, 2024 at 4:45 PM Michael Klishin ***@***.***>
wrote:

> There are also no reasons for the 5.17 users not to upgrade to 5.18 or
> 5.20. As stated earlier, this client usually ships minor releases without
> any breaking changes.
>
> Our small team has more important things to worry about than making sure a
> three minor behind patch release of this client has a backport accounted
> for in the NIST database.
>
> We have fixed the issue and released a round of back ports. NIST is
> notified. If someone has the cycle to update the NIST database, they can do
> it by providing evidence from this public repository.
>
> —
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_rabbitmq_rabbitmq-2Djava-2Dclient_issues_1062-23issuecomment-2D1953192248&d=DwMFaQ&c=udBTRvFvXC5Dhqg7UHpJlPps3mZ3LRxpb6__0PomBTQ&r=_bD89J3lxuba0TbAI8TNEY3DP0JXszUB1IfNbwFKo14&m=pX9Ep25BjgFAdUPCpnjgDljxHa7MMUTq1nGchR1K3ld28K80Ncw5rSKGaYN810nX&s=Ogm9H1OyDHBUhU674x1F8zLSRdRLhfpGW2iAiGxw-RY&e=>,
> or unsubscribe
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AWUHREEPD64D5P7FRRUR6GLYUPBV5AVCNFSM6AAAAAAZGA2VBCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSNJTGE4TEMRUHA&d=DwMFaQ&c=udBTRvFvXC5Dhqg7UHpJlPps3mZ3LRxpb6__0PomBTQ&r=_bD89J3lxuba0TbAI8TNEY3DP0JXszUB1IfNbwFKo14&m=pX9Ep25BjgFAdUPCpnjgDljxHa7MMUTq1nGchR1K3ld28K80Ncw5rSKGaYN810nX&s=dAkkpKMuK0lELHhr52GPDZPJL3Go6SaBAbUd7EG3QE4&e=>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
<http://www.verizon.com>

Peter Wolfe

Distinguished Engineer-Security Architect
Technology Platforms, Services, Solutions
TPD Product Security

M 732 514 2759

<https://www.verizon.com>

**Comment by michaelklishin** [comment]
@peterwvz we have done our job: we have responded to a responsibly disclosed vulnerability, patched it, backported it N times and produced a round of releases, then disclosed it when we were able to do so according to VMware vulnerability disclosure requirements.

That problem you have with a vendor on 5.14? Sorry, that's not our job. We are not paid by that vendor. In fact, that vendor might as well use open source RabbitMQ without a support contact and thus gets everything, including updates, for free, year after year.
